specs/004-enhance-the-casting/tasks.md

```
@@ -0,0 +1,1661 @@
+//! 
+//! - `char` → `u32`: Direct Unicode scalar value extraction
+//! - `u32` → `char`: Validated conversion (rejects surrogates U+D800-U+DFFF and values >U+10FFFF)
+//! - `char` ↔ other integers: Indirect via `u32`
+//! - `char` ↔ `String`: Runtime support required
+```
+**Dependencies**: T030  
+**Story**: US2
+
+---
+
+### T032: Update Documentation for Boolean/Character Conversions
```

markdownlint-cli2 (0.18.1)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)